### DISCUSSION TITLE
Custom CSS not propogating

### DISCUSSION THREAD

**Discussion by donbowen** [executablebooks/meta#590] [Q&A]
The default margin-bottom is 1.15rem for \<p\> elements on [my site](https://ledatascifi.github.io/ledatascifi-2022/content/02/10_Golden_3.html). I'm already using a custom CSS file, so I have the level of CSS skill that maximizes danger - I can do stuff, but don't fully understand why. 

The table in the page above has excess spacing, so I've tried to alter the custom css code to reduce the margin-bottom to 0. I've tried just about every combination between `p{margin-bottom:0}` and `.table p{margin-bottom:0}`, but nothing alters the style of p elements on the site. Both work as inline injections in the browser, but neither propogates from my custom css file.

**Comment by ZviBaratz** [comment]
Hi @donbowen, 
(not a maintainer, just a curious bystander :sweat_smile:)
Can you confirm any other CSS rules are applied successfully?
Also, did you try adding the [`!important`](https://www.w3schools.com/css/css_important.asp) tag?

**Comment by donbowen** [comment]
Hi and merry christmas! Thanks for chiming in.

Other CSS rules from the same css file are working clearly. For example,
the background color of the website.

I did try the !important tag even though some folks seem to hate its use.
But no luck.

On Sat, Dec 25, 2021 at 9:42 AM Zvi Baratz ***@***.***> wrote:

> Hi @donbowen <https://github.com/donbowen>,
> (not a maintainer, just a curious bystander 😅)
> Can you confirm any other CSS rules are applied successfully?
> Also, did you try adding the !important
> <https://www.w3schools.com/css/css_important.asp> tag?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/executablebooks/meta/discussions/590#discussioncomment-1869780>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMEHJ26GELYTCEJAWTS2GV3USXJ4VANCNFSM5KTT27CA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
---
Donald E. Bowen III

Assistant Professor of Finance
Perella Department of Finance, College of Business
Lehigh University
***@***.*** | ***@***.***
<412-326-9363>
https://bowen.finance

**Reply by choldgraf** [reply]
hmm strange - just adding a `p{margin-bottom: 0}` to the page via my chrome CSS seems to do the trick. Have you tried using the `inspect` tool via Chrome to confirm that the CSS rule is correctly selecting for the elements you want?

**Comment by ZviBaratz** [comment]
Sorry, I just now had a look. Really cool project! 
What about adding the `"mb-0"` class to the relevant tag?

**Comment by donbowen** [comment]
Clever idea. That would work for a one-off table fine, probably, but I
don't think it is a robust site-wide solution. My understanding is that I'd
have to replace all markdown tables I've written with raw html to add the
class to the tables.

**Comment by donbowen** [comment]
Same. I did the exact same to figure out the right tag and it works _in browser_ for me as well. In fact, I'd prefer `table p{margin-bottom: 0}` to restrict the change to tables only, which also works in the browser. But when I add it to my custom CSS and rebuild the site, nothing changes. It baffles me.

I should add that I'm using sphinx-book-theme 0.0.39 and jupyterbook 0.10.1 because I like the behavior of the TOC for 0.10 :)   I tried to migrate to 0.11 but updating the CSS to replicate my existing site design was a challenge and I saw indications that it might be impossible to do some things.

**Reply by ZviBaratz** [reply]
> I should add that I'm using sphinx-book-theme 0.0.39 and jupyterbook 0.10.1 because I like the behavior of the TOC for 0.10 :)

I second that. I can't seem to be able to create the TOC structure I'd like to have with the new format, either.

**Reply by choldgraf** [reply]
It'd be helpful if you'd be more specific about what you want to accomplish, and what result you're getting, to understand what has changed for the worse

**Reply by ZviBaratz** [reply]
@choldgraf the issues I'm facing are:

- I'd like to create a TOC containing both simple chapters (e.g. abstract and acknowledgements) and parts/chapters with sections (e.g. methods and results).
- I want to include a table of contents in the beginning of the book and specify the `max-depth` for it. It seems like that option is only available when using `parts`, but then I can't create the "simple" chapters (abstract, ack...).

I opened a discussion and gave a more detailed overview of the desired book structure [here](https://github.com/executablebooks/meta/discussions/592).
